File: src-1.x/BYTES.NET/WPF/Converters/BoolToVisibilityConverter.vb
'import .net namespace(s) needed
Imports System.Globalization

Namespace WPF.Converters

    ''' <summary>
    ''' converter returning the visibility state by boolean value
    ''' </summary>
    Public Class BoolToVisibilityConverter

        Implements IValueConverter

        Public Function Convert(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.Convert

            If IsNothing(parameter) OrElse CBool(parameter) Then 'return 'visible' on 'true'

                If CBool(value) Then

                    Return Visibility.Visible

                Else

                    Return Visibility.Collapsed

                End If

            Else

                If Not CBool(value) Then

                    Return Visibility.Visible

                Else

                    Return Visibility.Collapsed

                End If

            End If

        End Function

        Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.ConvertBack

            Return Binding.DoNothing

        End Function

    End Class

End Namespace

File: src-2023/BYTES.NET.WPF/Converters/BoolToVisibilityConverter.cs
//import .net namespace(s) required
using System;
using System.Collections.Generic;
using System.Globalization;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows;
using System.Windows.Data;

namespace BYTES.NET.WPF.Converters
{
    public class BoolToVisibilityConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
        {
            bool boolValue = (value is bool) ? (bool)value : false;
            bool boolParameter = (parameter is bool) ? (bool)parameter : true;

            if (boolParameter)
            {
                if (boolValue)
                {
                    return Visibility.Visible;
                }
                else
                {
                    return Visibility.Collapsed;
                }
            }
            else
            {
                if (!boolValue)
                {
                    return Visibility.Visible;
                }
                else
                {
                    return Visibility.Collapsed;
                }
            }
        }

        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
        {
            return Binding.DoNothing;
        }
    }
}
